app/scripts/controllers/serviceInstance.js

```diff
@@ -231,6 +235,17 @@ angular.module('openshiftConsole')
         };
       }));
 
+    $scope.$watch('isServiceInstanceReady(serviceInstance)', function(serviceInstanceReady) {
+      if (serviceInstanceReady) {
+        if ($scope.isMobileService($scope.serviceInstance) && $scope.isMobileEnabled) {
+          ServiceInstancesService.fetchServiceClassForInstance($scope.serviceInstance).then(function(serviceClass) {
```

Comment:
Does this need to be done when the `serviceInstance` is ready? Once we know it is:
```.js
if ($scope.isMobileService($scope.serviceInstance) && $scope.isMobileEnabled)
```
can we assume the serviceclass is needed?

If we are hiding it until we are ready we can use the filter in the html
```.html
<div ng-if="(serviceInstance | isServiceInstanceReady) && (serviceIntegrations | size)">
```

Reply:
@sedroche ServiceClass is needed as the integrations list is not available in the serviceInstance.

The serviceInstance needs to be ready to ensure that the mobile label has been added to the serviceInstance. However we can just check this by directly putting the watch on the isMobileService instead. Wdyt?